tests/Unit/Libraries/Cms/Updater/TufAdapterTest.php

```diff
@@ -0,0 +1,167 @@
+<?php
+
+/**
+ * @package     Joomla.UnitTest
+ * @subpackage  Updater
+ *
+ * @copyright   (C) 2024 Open Source Matters, Inc. <https://www.joomla.org>
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+namespace Joomla\Tests\Unit\Libraries\Cms\Updater;
+
+use Joomla\CMS\Updater\Adapter\TufAdapter;
+use Joomla\Tests\Unit\UnitTestCase;
+use Joomla\Utilities\ArrayHelper;
+use Tuf\Exception\MetadataException;
+
+/**
+ * Test class for Tuf Adapter.
+ *
+ * @package     Joomla.UnitTest
+ * @subpackage  Updater
+ * @since       __DEPLOY_VERSION__
+ */
+class TufAdapterTest extends UnitTestCase
+{
+    /**
+     * @return void
+     *
+     * @since   __DEPLOY_VERSION__
+     */
+    public function testProcessTufTargetThrowsExceptionIfHashesAreMissing()
+    {
+        $this->expectException(MetadataException::class);
+        $this->expectExceptionMessage("No trusted hashes are available for 'nohash.json'");
+
+        $object = $this->getMockBuilder(TufAdapter::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $method = $this->getPublicMethod($object, 'processTufTarget');
+        $method->invoke($object, 'nohash.json', []);
+    }
+
+    /**
+     * @return void
+     *
+     * @since   __DEPLOY_VERSION__
+     */
+    public function testProcesstuftargetAssignsCustomTargetKeys()
+    {
+        $object = $this->getMockBuilder(TufAdapter::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $method = $this->getPublicMethod($object, 'processTufTarget');
+        $result = $method->invoke($object, 'targets.json', $this->getMockTarget([
+            'custom' => [
+                'name'    => 'Testupdate',
+                'version' => '1.2.3',
+            ],
+        ]));
+
+        $this->assertSame('Testupdate', $result['name']);
+        $this->assertSame('1.2.3', $result['version']);
+    }
+
+    /**
+     * @return void
+     *
+     * @since   __DEPLOY_VERSION__
+     */
+    public function testProcesstuftargetAssignsClientId()
+    {
+        $object = $this->getMockBuilder(TufAdapter::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+
+        $method = $this->getPublicMethod($object, 'processTufTarget');
+        $result = $method->invoke($object, 'targets.json', $this->getMockTarget([
+            'client' => 'site',
+        ]));
+
+        $this->assertSame(0, $result['client']);
```

Comment:
```suggestion
        $this->assertSame(0, $result['client_id']);
```
Needs to be adapted to latest change for client_id.